### PR TITLE
[REEF-489] Add concept of application attempt to driver startup

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Adds a getter to resubmissionAttempts for DriverRestarted and IDriverRestarted.
  * Modify the DriverRestart example to check for only restarted once.

JIRA:
  [REEF-489](https://issues.apache.org/jira/browse/REEF-489)

### CODE REVIEW

**Comment by afchung** [comment]
@beysims Please have a look at the bridge portion. Currently testing on HDInsight.

**Comment by afchung** [comment]
Tests pass on HDInsight.

**Comment by jwang98052** [comment]
I will review it.

File: lang/cs/Org.Apache.REEF.Driver/IDriverRestarted.cs [hunk 1]
```diff
@@ -31,6 +31,11 @@ public interface IDriverRestarted : IDriverStarted
         /// The set of expected Evaluator IDs that are returned to the Driver by the
         /// RM on Driver Restart.
         /// </summary>
-        ISet<string> ExpectedEvaluatorIds { get; } 
+        ISet<string> ExpectedEvaluatorIds { get; }
+
+        /// <summary>
+        /// The number of times the Driver has been resubmitted. Does not include the initial submission.
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Driver/IDriverRestarted.cs]: Please clarify. It will be `0` on first launch, `1` for the first restart, ...?

Comment (afchung) [inline lang/cs/Org.Apache.REEF.Driver/IDriverRestarted.cs]: Yes, that is right.

**Comment by afchung** [comment]
Seems like there's a merge conflict. I'll fix and update.

**Comment by afchung** [comment]
@markusweimer, @jwang98052 Fixed merge. Please continue your review, and I'll fix the changes in one shot after your pings. Thanks!

**Comment by afchung** [comment]
I was just notified of @beysims 's change. Since this PR involves changing some bridge code, I'll make sure it passes on HDInsight first. Thanks!

**Comment by afchung** [comment]
@markusweimer @jwang98052 Reopened as #501.

**Pull request closed without merging**